packages/react-components/react-search/src/components/SearchBox/renderSearchBox.tsx

```diff
@@ -15,15 +15,17 @@ export const renderSearchBox_unstable = (state: SearchBoxState) => {
 
   // TODO Add additional slots in the appropriate place
   const rootSlots = {
-    contentAfter: slots.contentAfter && {
-      ...slotProps.contentAfter,
-      children: (
-        <>
-          {slotProps.contentAfter.children}
-          {slots.dismiss && <slots.dismiss {...slotProps.dismiss} />}
-        </>
-      ),
-    },
+    contentAfter:
+      (slots.contentAfter && {
+        ...slotProps.contentAfter,
+        children: (
+          <>
+            {slotProps.contentAfter.children}
+            {slots.dismiss && <slots.dismiss {...slotProps.dismiss} />}
+          </>
+        ),
+      }) ||
+      (slots.dismiss && <slots.dismiss {...slotProps.dismiss} />),
```

Comment:
I'd lean towards saying it's OK for the dismiss slot to go away if someone explicitly does `contentAfter={null}`, but I'm open to reasons indicating otherwise